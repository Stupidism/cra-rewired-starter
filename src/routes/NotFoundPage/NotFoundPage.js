import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <h1>
    Page Not found. Go <Link to="/">Home</Link>
  </h1>
);

export default NotFoundPage;
