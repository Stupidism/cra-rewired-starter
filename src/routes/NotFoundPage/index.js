import createAsyncComponent from 'modules/createAsyncComponent';

export default createAsyncComponent(() =>
  import('./NotFoundPage' /* webpackChunkName: "routes.NotFoundPage" */),
);
