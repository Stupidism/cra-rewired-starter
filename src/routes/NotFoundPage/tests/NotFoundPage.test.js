import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Link } from 'react-router-dom';

import NotFoundPage from '../NotFoundPage';

describe('<App />', () => {
  it('should be consistent with snapshot', () => {
    const snapshot = mount(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>,
    );
    expect(snapshot).toMatchSnapshot();
  });

  it('should render a `Link` to homepage', () => {
    const wrapper = mount(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>,
    );
    expect(wrapper.find(Link)).toHaveLength(1);
    expect(
      wrapper
        .find(Link)
        .getDOMNode()
        .getAttribute('href'),
    ).toBe('/');
    expect(wrapper.find(Link).text()).toBe('Home');
  });
});
