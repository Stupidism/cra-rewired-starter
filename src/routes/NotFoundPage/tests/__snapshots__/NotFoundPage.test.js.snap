// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should be consistent with snapshot 1`] = `
<h1>
  Page Not found. Go 
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</h1>
`;
